pool/src/pool.rs

```diff
@@ -159,6 +163,50 @@ impl Pool {
 		Ok(Some(tx))
 	}
 
+	/// evict a transaction from the pool
+	pub fn evict_from_pool(&mut self) {
+		// Check that the new tx is not a bullshit tx in term of fee
+
+		let mut ordered_entries = self.entries.clone();
```

Comment:
This cloning makes me uncomfortable. I understand that's how the pool operates generally (cloning a lot) so I think it's fine for this PR. But we should track this and other entries cloning in another issue to optimize all that. It's a lot of allocation and copying (including range proofs, etc) just to order based on weight.

Reply:
Oups I forgot a comment there ^^. Also yeah definitely agree that this cloning is quite heavy in term of ressources.

Reply:
Also I missing a check where I compare the tx fee/weight ratio with the ones in the pool so we don't remove a higher fee transaction for a low fee one.